packages/clay-form/src/InputWithAutocomplete.tsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayAutocomplete from '@clayui/autocomplete';
+import ClayDropDown from '@clayui/drop-down';
+import React, {useEffect, useRef, useState} from 'react';
+import {useKeyHandlerForList} from '@clayui/shared';
+
+interface IProps extends React.InputHTMLAttributes<HTMLInputElement> {
+	/**
+	 * Flag to indicate is loading indicator should be shown.
+	 */
+	loading?: boolean;
+
+	/**
+	 * Items that will show up in the dropdown list
+	 */
+	items: Array<string> | Array<Object>;
+
+	/**
+	 * Function that returns the value that should be showed in the list.
+	 */
+	itemSelector?: (val: string | Object) => string;
+
+	/**
+	 * Callback for when input value changes
+	 */
+	onItemSelect: (val: string | Object) => string | Object;
+
+	/**
+	 * Callback for when input value changes
+	 */
+	onValueChange: (val: string | Object) => void;
+}
+
+export const ClayInputWithAutocomplete: React.FunctionComponent<IProps> = ({
+	items = [],
+	itemSelector = val => val as string,
+	loading,
+	onItemSelect,
+	onValueChange,
+	value = '',
+	...otherProps
+}) => {
+	const activeListItemRef = useRef<HTMLLIElement>(null);
+	const inputRef = useRef<HTMLInputElement>(null);
+	const [activeIndex, setActiveIndex] = useState(-1);
+	const [active, setActive] = useState(!!value);
+
+	const keyHandler = useKeyHandlerForList({
+		activeListItemRef,
+		index: activeIndex,
+		inputRef,
+		onIndexChange: setActiveIndex,
+		onIndexSelect: (index: number) => onItemSelect(items[index]),
+		totalItems: items.length,
+	});
+
+	useEffect(() => {
+		setActive(!!value);
+	}, [value]);
+
+	useEffect(() => {
+		setActiveIndex(-1);
+	}, [active]);
+
+	return (
+		<ClayAutocomplete>
+			<ClayAutocomplete.Input
+				{...otherProps}
+				onChange={e => onValueChange(e.target.value)}
+				onKeyDown={keyHandler}
+				ref={inputRef}
+				value={value}
+			/>
+
+			<ClayAutocomplete.DropDown active={active} onSetActive={setActive}>
+				<ClayDropDown.ItemList>
+					{items.map((item, i) => (
+						<ClayAutocomplete.Item
+							active={i === activeIndex}
+							key={itemSelector(item)}
+							match={value.toString()}
+							onClick={() => onItemSelect(itemSelector(item))}
+							ref={
+								activeIndex === i
+									? activeListItemRef
+									: undefined
+							}
+							value={itemSelector(item)}
+						/>
+					))}
+				</ClayDropDown.ItemList>
+			</ClayAutocomplete.DropDown>
+
+			{loading && <ClayAutocomplete.LoadingIndicator />}
```

Comment:
Another thing that has to enter the high levels is when loading is enabled, we should also show in the DropDown the message of `Loading...` as in the example of stories of autocomplete.

This is the additional document to handle the use cases of loading. https://docs.google.com/document/d/1ZkpPajJBLioHBd2sEwE7Sb0FPdYr9ypwdL2ebHed1aQ/edit?ts=5c659b11

Reply:
Is there any way to centralize these docs or make them more findable? I have no idea how to find them currently, even when I search google drive nothing shows up.

Could we link them to tickets?

Reply:
Yes that's a bit chaotic, sometimes I do not find the documents either... I just had access because of the LEXI tickets they added. I think it would be best to have such documents in github 🙂.